### PR TITLE
Add missing 13_callback.core docments

### LINKED ISSUE #34: 13_callback.core: create docments
Add docments to [13_callback.core](https://github.com/muellerzr/fastai-docment-sprint/blob/master/nbs/13_callback.core.ipynb).

See the style guide and contribution guide for more details (links will be added when created).

This requires moderate to advanced familiarity with the fastai training loop and callbacks.

Docments progress:

- [x] Callback
- [x] TrainEvalCallback
- [x] Gather and fetch preds callbacks

### PR DESCRIPTION
**Pull request by warner-benjamin**
This should resolve #34 by adding the rest of the documentation. Also contains minor edits to #74.

I decided to add a return type hint for a list of 2-4 tensors in the format `all_tensors(self) -> (Tensor, list)`, but can change if we want it to have a different format.

### CODE REVIEW

File: fastai/callback/core.py [hunk 1]
```diff
@@ -165,15 +166,15 @@ def __init__(self,
         ds_idx:int=1, # Index of dataset, 0 for train, 1 for valid, used if `dl` is not present
         dl:DataLoader=None, # `DataLoader` used for fetching `Learner` predictions
         with_input:bool=False, # Whether to return inputs in `GatherPredsCallback`
-        with_decoded:bool=False, # Whether to return predicted classes
-        cbs:list=None, # `Callback` list to add to the `Learner`
+        with_decoded:bool=False, # Whether to return decoded predictions
+        cbs:(Callback,list)=None, # `Callback` to temporarily remove from `Learner`
```
Comment (muellerzr) [inline fastai/callback/core.py:170]:
```suggestion
        cbs:(Callback,list)=None, # `Callback`(s) to temporarily remove from `Learner`
```

**Review by muellerzr — APPROVED** [review]
Nicely done. *Super* tiny nit 😄

File: fastai/callback/core.py (hunk 1 above)
Comment (warner-benjamin) [reply fastai/callback/core.py:170]: I think the (s) after the hotlink looks a bit silly, which is why I left it off.

**Review by marii-moe — COMMENTED** [review]
Left a comment, I expect you had a good reason for changing `with_decoded`, so I am good with approving if that is explained.

File: fastai/callback/core.py [hunk 2]
```diff
@@ -165,15 +166,15 @@ def __init__(self,
         ds_idx:int=1, # Index of dataset, 0 for train, 1 for valid, used if `dl` is not present
         dl:DataLoader=None, # `DataLoader` used for fetching `Learner` predictions
         with_input:bool=False, # Whether to return inputs in `GatherPredsCallback`
-        with_decoded:bool=False, # Whether to return predicted classes
-        cbs:list=None, # `Callback` list to add to the `Learner`
+        with_decoded:bool=False, # Whether to return decoded predictions
```
[the remark this answers is not in the document]
Comment (warner-benjamin) [reply fastai/callback/core.py:169]: I thought predicted classes implied a classification problem and decoded predictions was a more accurate description as decoding could be unstandardizing regression results, for example.

[the remark this answers is not in the document]
Comment (marii-moe) [reply fastai/callback/core.py:169]: I forgot about regression, and was thinking about how most things use classes in one way or another. Good call.

**Review by marii-moe — APPROVED** [review]
LGTM

**Pull request merged**